cg/meta/archive/archive.py

```diff
@@ -0,0 +1,147 @@
+        files_to_retrieve: List[TransferData] = []
+        for sample_id in sample_ids:
+            files_to_retrieve.extend(
+                TransferData(destination=sample_id, source=file.as_posix())
+                for file in self.housekeeper_api.get_non_archived_files(
+                    bundle_name=sample_id, tags=[SequencingFileTag.SPRING]
+                )
+            )
+        retrieve_task_id: int = self.ddn_api.retrieve_folders(
+            sources_and_destinations=files_to_retrieve
+        )
+        self.housekeeper_api.add_retrieve_task(
+            files=[Path(transfer_data.source) for transfer_data in files_to_retrieve],
+            retrieve_task_id=retrieve_task_id,
+        )
+
+    def archive_all_non_archived_spring_files(self) -> None:
+        """Archives all non archived spring files (so far only for DDN customers)."""
+
+        spring_files_to_archive: List[str] = [
+            file.path for file in self.housekeeper_api.get_all_non_archived_spring_files()
+        ]
+
+        if not spring_files_to_archive:
+            LOG.info("Found no files ready to be archived.")
+            return
+
+        sample_and_spring_files_per_archive_location: Dict[
+            str, List[SampleAndFile]
+        ] = self.sort_spring_files_on_archive_location(spring_files_to_archive)
+        files_to_archive: List[TransferData] = [
+            TransferData(destination=ddn_file_to_archive.sample, source=ddn_file_to_archive.file)
+            for ddn_file_to_archive in sample_and_spring_files_per_archive_location[DDN]
+        ]
+        archive_task_id: int = self.ddn_api.archive_folders(
+            sources_and_destinations=files_to_archive
+        )
+        self.housekeeper_api.add_archives(
+            files=[Path(transfer_data.source) for transfer_data in files_to_archive],
+            archive_task_id=archive_task_id,
+        )
+
+    def sort_spring_files_on_archive_location(
+        self, files: List[str]
+    ) -> Dict[str, List[SampleAndFile]]:
+        """Sort the given list of files and gives back a dictionary
+        mapping the data archive location to a tuple containing sample id and file path."""
+        files_per_archive_location: Dict[str, List[SampleAndFile]] = {}
+        for file in files:
+            location_and_sample: ArchiveLocationAndSample = (
+                self.get_archive_location_and_sample_id_from_file_path(file)
+            )
+            if not location_and_sample:
+                continue
+            archive_location: str = location_and_sample.location
+            sample_id: str = location_and_sample.sample_id
+            if files_per_archive_location.get(location_and_sample.location):
+                files_per_archive_location[archive_location].append(SampleAndFile(sample_id, file))
+            else:
+                files_per_archive_location[archive_location] = [SampleAndFile(sample_id, file)]
+        return files_per_archive_location
```

Comment:
I think this logic is smart in a future where we archive all files. However currently we have 370 000 spring files, and for each we will query status_db for a sample. I think this is fine since, we will archive everything in the future, but we need to make sure this function doesn't take 15 min to run.